src/kubernetes.cc

```diff
@@ -356,7 +381,10 @@ KubernetesReader::GetPodAndContainerMetadata(
                              collected_at, is_deleted));
   }
 
-  result.emplace_back(GetPodMetadata(pod, collected_at, is_deleted));
+  auto cb = [=](const json::Object* pod) {
+    return PodResourceMappingCallback(pod);
+  };
+  result.emplace_back(GetObjectMetadata(pod, collected_at, is_deleted, cb));
```

Comment:
I did realize that you can have `GetObjectMetadata` return _just_ the metadata, and compose that into `ResourceMetadata` here, but I'm not going to block this PR on that. We can refactor it later.
For the record, I'm proposing something like this:
```c++
    IdsAndMR ids_and_mr = PodResourceMappingCallback(object);
    result.emplace_back(
        ids_and_mr.first, ids_and_mr.second,
        std::move(GetObjectMetadata(pod, collected_at, is_deleted)));
```
Note that in this scenario `PodResourceMappingCallback` is no longer a callback, so should probably be renamed to `PodResourceMapping`. Ditto for nodes, etc.
In `ObjectWatchCallback`, this literally becomes:
```c++
    IdsAndMR ids_and_mr = resource_mapping_cb(object);
    result.emplace_back(
        ids_and_mr.first, ids_and_mr.second,
        std::move(GetObjectMetadata(pod, collected_at, is_deleted)));
```

Reply:
lets do this in a separate PR. Thanks for writing it up in so much detail.